cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -377,12 +375,14 @@ static int _sc_cgroup_v2_init_bpf(sc_device_cgroup *self, int flags) {
     }
 
     if (!from_existing) {
-        /* 4.5+ kernels allow maps to be created by simple users otherwise the
-         * calls fail with EPERM */
+        /* load and attach the BPF program as root */
+        (void)sc_set_effective_identity(sc_root_group_identity());
         int prog_fd = load_devcgroup_prog(devmap_fd);
-        if (bpf_prog_attach(BPF_CGROUP_DEVICE, cgroup_fd, prog_fd) < 0) {
+        int attach = bpf_prog_attach(BPF_CGROUP_DEVICE, cgroup_fd, prog_fd);
+        if (attach < 0) {
             die("cannot attach cgroup program");
         }
+        (void)sc_set_effective_identity(old);
```

Comment:
Should we ignore an error here on dropping privileges back to the unprivileged user? Otherwise we continue to execute as root which could be bad..?

Reply:
sc_set_effective_identity() dies inside when setegid/seteuid fail. The only returned thing are the previous euid/egid values.

Reply:
Ah thanks I missed that - great!

Reply:
Just in case it is not clear, +1 from me for security review.